### ISSUE TITLE
I want Mac support...

### ISSUE THREAD

**Issue by kennym** [matiasinsaurralde/facebook-tunnel#1]
...because I am an Apple user. ;)

**Comment by ryanjamurphy** [comment]
@kennym any luck on this? any way I can help?

**Comment by cookietime--yay** [comment]
I tried to replace it with ```#include <net/if.h>``` . I'm not sure if that is a good solution but the error is gone. The next error it shows is:
```
/Users/.../facebook-tunnel/tun.cpp:43:10: fatal error: 
      'linux/if_tun.h' file not found
#include <linux/if_tun.h>
         ^
1 error generated.
make[2]: *** [CMakeFiles/tun.dir/tun.cpp.o] Error 1
make[1]: *** [CMakeFiles/tun.dir/all] Error 2
make: *** [all] Error 2 
```

After googling around i guess Mac does not have ```if_tun.h``` in ```/usr/include/net/```, so I'm stuck here.
Maybe someone should just rewrite ```tun.cpp``` and ```tun.c``` for Mac ...

**Comment by kennym** [comment]
> After googling around i guess Mac does not have if_tun.h in /usr/include/net/, so I'm stuck here.
Maybe someone should just rewrite tun.cpp and tun.c for Mac ...
That's what is blocking me on Mac. Anyone smart enough to come up with an implementation or alternative?

**Comment by TalentFang** [comment]
Hi kennym:
  is there any solution found for this problem? thanks.

**Comment by theGowda** [comment]
> http://tuntaposx.sourceforge.net/

were you able to find a solution?

**Comment by kennym** [comment]
Nope, and I'm pretty sure Facebook has changed their API by a lot by now, but please try.

**Comment by theGowda** [comment]
Ya I was not concerned about facebook tunnel specifically. I was just trying to port an old code base written for linux to mac os. And this if.h was the only thing remaining. So wanted to know how you all fixed it. 
Cheers!
